### PR TITLE
Fix snow git execute for Python files

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Fix `snow git execute` for Python files:
1. Copy the files first to temporary stage to workaround missing Snow git feature
2. Strip leading and trailing quotes from input variables so the values are consistent between SQL and Python
3. Add support for temporary stage in `StageManager`

### CODE REVIEW

**Review by sfc-gh-astus — DISMISSED** [review]

File: tests/git/test_git_commands.py [hunk 1]
```diff
@@ -587,20 +589,29 @@ def test_execute(
 ):
     mock_execute.return_value = mock_cursor(
         [
-            {"name": "repo/branches/main/a/S3.sql"},
-            {"name": "repo/branches/main/s1.sql"},
-            {"name": "repo/branches/main/s2"},
+            {"name": "/a/S3.sql"},
+            {"name": "/s1.sql"},
+            {"name": "/s2"},
```
Comment (sfc-gh-astus) [inline tests/git/test_git_commands.py:594]: Why you changed list-files response?

File: tests/git/test_git_commands.py [hunk 2]
```diff
@@ -587,20 +589,29 @@ def test_execute(
...
     assert result.exit_code == 0, result.output
-    ls_call, *execute_calls = mock_execute.mock_calls
-    assert ls_call == mock.call(f"ls {expected_stage}", cursor_class=DictCursor)
+    create_call, copy_call, ls_call, *execute_calls = mock_execute.mock_calls
+    assert create_call == mock.call(
+        "create temporary stage if not exists IDENTIFIER('FOO.BAR.snowflake_cli_tmp_stage_123')"
+    )
+    assert copy_call == mock.call(
+        f"copy files into @FOO.BAR.snowflake_cli_tmp_stage_123/ from {expected_stage}"
+    )
```
Comment (sfc-gh-astus) [inline tests/git/test_git_commands.py:608]: Why you copy sql files to temporary stage? They don't need it.

File: tests/git/test_git_commands.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply tests/git/test_git_commands.py:594]: Because the list is done on stage not on repo.

File: tests/git/test_git_commands.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply tests/git/test_git_commands.py:608]: I'm copying git contents to stage. I don't differentiate if this is SQL or Python.

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**